source/specify-a-query.txt

```diff
@@ -0,0 +1,266 @@
+.. _pymongo-specify-query:
+
+===============
+Specify a Query
+===============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: expressions, operations, read, write, filter
+
+Overview
+--------
+
+In this guide, you can learn how to specify a query by using {+driver-short+}.
+
+You can refine the set of documents that a query returns by creating a
+**query filter**. A query filter is an expression that specifies the search
+criteria MongoDB uses to match documents in a read or write operation.
+In a query filter, you can prompt the driver to search for documents with an
+exact match to your query, or you can compose query filters to express more
+complex matching criteria.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the following documents in a collection called
+``fruits``:
+
+.. code-block:: json
+
+   { "_id": 1, "name": "apples", "qty": 5, "rating": 3, "color": "red", "type": ["fuji", "honeycrisp"] },
+   { "_id": 2, "name": "bananas", "qty": 7, "rating": 4, "color": "yellow", "type": ["cavendish"] },
+   { "_id": 3, "name": "oranges", "qty": 6, "rating": 2, "type": ["naval", "mandarin"] },
+   { "_id": 4, "name": "pineapple", "qty": 3, "rating": 5, "color": "yellow" },
+
+The following code example shows how to create a database and collection, and
+insert the sample documents into your collection:
+
+.. code-block:: python
+
+   from pymongo import MongoClient
+
+   uri = "<Your Connection URI>"
+   client = MongoClient(uri)
+
+   try:
+       database = client["sample_fruit"]
+       collection = database["fruits"]
+
+       collection.insert_many([
+           { "_id": 1, "name": "apples", "qty": 5, "rating": 3, "color": "red", "type": ["fuji", "honeycrisp"] },
+           { "_id": 2, "name": "bananas", "qty": 7, "rating": 4, "color": "yellow", "type": ["cavendish"] },
+           { "_id": 3, "name": "oranges", "qty": 6, "rating": 2, "type": ["naval", "mandarin"] },
+           { "_id": 4, "name": "pineapple", "qty": 3, "rating": 5, "color": "yellow" },
+       ])
+
+       client.close()
+
+   except Exception as e:
+      raise Exception("Unable to insert the documents due to the following error: ", e)
+
+Exact Match
+-----------
+
+Literal value queries return documents with an exact match to your query filter.
+
+The following example specifies a query filter as a parameter to the ``find()``
+method. The method finds all documents with a ``color`` field value of ``"yellow"``:
+
+.. code-block::
+   :copyable:
+
+   results = collection.find({ "color": "yellow" })
```

Comment:
S: include output for all code examples as done [here](https://www.mongodb.com/docs/drivers/rust/current/fundamentals/crud/read-operations/query/)

Reply:
Included in all but the "find all" note (since it shows all the documents right before that)